Assets/Scripts/Player/PlayerMovement.cs

```diff
@@ -15,34 +15,72 @@ public class PlayerMovement : MonoBehaviour
     private Vector3 movement;
     private CharacterController controller;
 
+    private Camera cam;
+
     private void Awake()
     {
         controller = GetComponent<CharacterController>();
+
+        // TC: cache the camera reference
+        cam = Camera.main;
     }
 
     //read movement inputs
     private void OnMove(InputValue value)
     {
-        movement.x = value.Get<Vector2>().x;
-        movement.z = value.Get<Vector2>().y;
-        movement = movement.normalized;
+        // TC: store the movement vector in one single get and normalize call
+        movement = value.Get<Vector2>().normalized;
+
+        //movement.x = value.Get<Vector2>().x;
+        //movement.z = value.Get<Vector2>().y;
+        //movement = movement.normalized;
     }
 
     //move
     private void Update()
     {
-        //Apply movement
-        AlignMovement(movement);
-        controller.Move(new Vector3(movement.x, 0f, movement.z) * moveSpeed * Time.deltaTime);
+        var finalMovement = CalculateMovementVector();
```

Comment:
created a new function to calculate a `finalMovement` Vector3 which is based on where the camera is facing and the player input